src/components/actions/helpers.ts

```diff
@@ -44,7 +46,60 @@ export const swapOutAliasedActions = async (templateId: number, action: ActionIn
   if (altCondition !== undefined) aliasedAction.condition = altCondition
 
   // Override parameters
-  aliasedAction.parameter_queries = { ...aliasedAction.parameter_queries, ...overrideParams }
+  aliasedAction.parameter_queries = merge(aliasedAction.parameter_queries, overrideParams)
 
   return aliasedAction
 }
+
+interface ActionResult {
```

Comment:
hmm maybe `processTrigger(..): ActionResult[]`

Reply:
Yup, you mean adding return type to processTrigger:
```
export async function processTrigger(payload: TriggerPayload): Promise<ActionResult[]> {
...
```

Moved "ActionResult" to a shared type and added it to processTrigger: 96294828